### PR TITLE
[Web-29] Custom Course Menu

### PR DESCRIPTION
**Pull request by emilyPhee**
https://user-images.githubusercontent.com/57051508/220432479-3474f4ea-09fc-4e25-a201-788934fea4fc.mov

- Use Chakra UI Popover and have custom course input fields
  - validation for inputs 
- Show AppToast when adding course item
 
*fixed AppToast style

### CODE REVIEW

**Comment by emilyPhee** [comment]
[[Web] Add menu to Custom course](https://trello.com/c/1zRne6Gb/29-web-add-menu-to-custom-course)

File: new-nextjs-client/src/components/index-page/CourseSearchWindow/AddCustomCourse/AddCustomCourse.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+          Custom
+        </AppButton>
+      </PopoverTrigger>
+      <PopoverContent
+        borderRadius={borderRadiusSM}
+        borderColor={theme === 'light' ? gray5 : gray4}
+        color={theme === 'light' ? defaultText : defaultTextDark}
+        padding="1.2rem .8rem"
+        bg={theme === 'light' ? gray7 : gray2}
+        // bg={theme === 'light' ? bgColor2 : bgColorDark2}
```
Comment (anthopark) [inline new-nextjs-client/src/components/index-page/CourseSearchWindow/AddCustomCourse/AddCustomCourse.tsx]: Don't leave commented code

File: new-nextjs-client/src/components/shared/AppInput/AppInput.tsx [hunk 2]
```diff
@@ -20,16 +20,18 @@ import {
   gray2,
 } from '@styles/variables';
 import { useTheme } from 'next-themes';
-import { useEffect, useState } from 'react';
+import { forwardRef, useEffect, useState } from 'react';
 
 interface Props {
   onChange?: (newValue: any) => void;
   value?: any;
+  id?: string | undefined;
```
Comment (anthopark) [inline new-nextjs-client/src/components/shared/AppInput/AppInput.tsx]: You can just leave `id:? string;`. this implies `id` can be `undefined`

File: new-nextjs-client/src/styles/variables.ts [hunk 3]
```diff
@@ -27,6 +27,7 @@ const blue2 = '#397BDD';
 /**************** 
   Control Sizes (Input, Button, etc.)
 ****************/
+const controlHeightSM = '3.4rem';
```
Comment (anthopark) [inline new-nextjs-client/src/styles/variables.ts:30]: Nice job on using the variable for this!

**Review by anthopark — APPROVED** [review]
Looks good. just some minor stuff!

File: new-nextjs-client/src/styles/variables.ts (hunk 3 above)
Comment (emilyPhee) [reply new-nextjs-client/src/styles/variables.ts:30]: thank you! 😄

**Pull request merged**